core/v1alpha1/federationstatus_types.go

```diff
@@ -0,0 +1,174 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories=maistra-io
+
+// FederationStatus is the Schema for reporting the status of a MeshFederation.
+// The name of the FederationStatus will match the name of the istiod pod to
+// which it is associated.  There will be one FederationStatus resource for
+// each istiod pod.
+type FederationStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is unused
+	Spec FederationStatusSpec `json:"spec,omitempty"`
+	// Status of the mesh federations
+	Status FederationStatusStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FederationStatusList contains a list of FederationStatus
+type FederationStatusList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []FederationStatus `json:"items"`
+}
+
+// FederationStatusSpec is always empty
+type FederationStatusSpec struct{}
+
+// FederationStatusStatus represents the status of each federated mesh.
+type FederationStatusStatus struct {
+	// Meshes represents the status of each federated mesh.  The key
+	// represents the name of the MeshFederation resource configuring
+	// federation for with a remote mesh (as namespace/name).
+	// +optional
+	// +nullable
+	// +patchMergeKey=mesh
+	// +patchStrategy=merge,retainKeys
+	Meshes []FederationStatusDetails `json:"meshes,omitempty"`
```

Comment:
Was this supposed to be a map?

Reply:
Maps don't work very well in k8s resource types (try finding an example in k8s).  Notice the +patch* tags.  The list is a map, where the `mesh` field acts as the key.  Same will be true for `remotes`, where the `source` field acts as a key.